### DISCUSSION TITLE
How do I get the output of a program

### DISCUSSION THREAD

**Discussion by greg0ire** [davatorium/rofi#2018] [(Possible) Issues]
I'm trying to run Signal from rofi. When I run it from a terminal, it works great. When I run rofi from a terminal, and then run Signal from rofi, it works great as well. But when I try running Signal from rofi, launching it via a shortcut defined in my Sway config, it doesn't work. How can I get `stdout` and `stderr` from Signal? I've tried creating an intermediate shell script, but it makes the bug disappear. I suspect it might have to do with the environment, so alternatively, how do I find what the environment is?

**Comment by DaveDavenport** [comment]
The environment is the same as it is for your sway instance. 
Maybe bake a  binding for `rofi -show env -modes env:/usr/bin/env`  to see the environment?

**Reply by greg0ire** [reply]
Great idea! It displays an env that I can also send to a file, like this: `bindsym $mod+o exec /usr/bin/env > /tmp/some_file`. Next, I tried binding that shortcut to `signal-desktop`, thinking it wouldn't work because the env variables are almost identical and… but to my surprise, Signal worked. I think that means it's not really an issue with the environment provided by sway to rofi then?!


# Shortcut env

```
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
DEBUGINFOD_URLS=https://debuginfod.archlinux.org
DESKTOP_SESSION=sway
DESKTOP_STARTUP_ID=c08d7d5eab617afcd1cfc4b956bb807f
DISPLAY=:1
HOME=/home/gregoire
I3SOCK=/run/user/1000/sway-ipc.1000.1755.sock
LANG=en_US.UTF-8
LOGNAME=gregoire
MAIL=/var/spool/mail/gregoire
MOTD_SHOWN=pam
OLDPWD=/home/gregoire
PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
PWD=/home/gregoire
SHELL=/usr/bin/zsh
SHLVL=1
SWAYSOCK=/run/user/1000/sway-ipc.1000.1755.sock
USER=gregoire
WAYLAND_DISPLAY=wayland-1
XCURSOR_SIZE=24
XDG_ACTIVATION_TOKEN=c08d7d5eab617afcd1cfc4b956bb807f
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_SESSION_CLASS=user
XDG_SESSION_DESKTOP=
XDG_SESSION_ID=2
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session1
XDG_SESSION_TYPE=wayland
XDG_VTNR=1
_=/usr/bin/env
```

# Rofi env

![2024-08-09_10-01](https://github.com/user-attachments/assets/cce49818-5b14-47ff-87d8-d2d851933ae9)

![2024-08-09_09-57](https://github.com/user-attachments/assets/97a46c80-f596-49e7-b739-67f7b987f322)
![2024-08-09_09-59](https://github.com/user-attachments/assets/316fe719-256f-4a63-97d6-d0cf2faac647)

**Reply by greg0ire** [reply]
I tried something else. This: `set $menu rofi -show run -run-command {cmd}`. For some reason, signal works. I found out because I was trying  `set $menu rofi -show run -run-command {cmd} > /tmp/output 2>&1` and that worked as well. I really don't get it.
